etc/afpd/afp_config.c

```diff
@@ -47,6 +48,61 @@
 #include "uam_auth.h"
 #include "status.h"
 #include "volume.h"
+#include "dircache.h"
+
+/*!
+ * Safe string to integer conversion with bounds checking
+ *
+ * @param str         String to convert
+ * @param param_name  Parameter name for error messages
+ * @param min_val     Minimum allowed value
+ * @param max_val     Maximum allowed value
+ * @param default_val Default value to return on error
+ *
+ * @returns Converted integer value or default_val on error
+ */
+static int safe_atoi(const char *str, const char *param_name,
```

Comment:
A similar function exists in lantest.c – can a case be made that this should live in libatalk (e.g. util module)?

Reply:
I have been thinking the same thing :)

Reply:
I couldn't find an obvious existing source file to put it in so perhaps a brand new one makes sense.

Reply:
Yea lets save it for another PR. This one is already very big (commit 1=new feature + commit 2=add missing strict dircache cleanup + commit 3=significant legacy bugfix for ACL test failures!).
I want to sort out the DSI usage in the testsuite, so I can create a common use safe_atoi at the same time as part of a clean-up PR.

That ACL bug was really tough to nail down! It was a pointer corruption bug in uuid.c:getuuidfromname() which was resulting in calling getpwnam with root, instead of the user! Thankfully the code erred instead of allowing an elevation. I suspect this may be the same issue I have seen on FreeBSD with ZFS ACLs where you cannot delete some files randomly.

Reply:
@rdmark PS; You can now disable `-Dwith-spotlight` in the Alpine container if you want :)